### ISSUE TITLE
Allow specifying raw file name

### ISSUE THREAD

**Issue by ansman** [cashapp/paparazzi#1085]
I run my tests based on java resource files using a parameterized test. So all my tests run from a single method.

This is problematic because Paparazzi will prepend the test package name, class name and method name to the output file. To make matters worse, the output file name is often too long since I use the FQN of the composable being screenshotted.

Specifying a name doesn't work as it only appends it to the default name.

There should be a way to specify the full file name without having the class name as a prefix.

Here is some code to illustrate the problem:
```kotlin
@RunWith(Parameterized::class)
class PaparazziScreenshotTests(private val test: TestCase) {
    @get:Rule
    val paparazzi = Paparazzi(...)

    @Test
    fun runTest() {
        paparazzi.snapshot {
            ...
        }
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters(name = "{0}")
        fun testCases() = listResources().map(::TestCase)
    }

    class TestCase(
        val className: String,
        val methodName: String,
    ) {
        override fun toString(): String = buildString {
            append(className)
            append('.')
            append(methodName)
        }
    }
}
```
This will generate files that look like this:
```
com.example_PaparazziScreenshotTests_runTest[com.example.SomeComposableKt.SomeComposable].png
```

**Comment by ansman** [comment]
I'm happy to provide a PR that either allows disabling the prepending or a way to specify the full filename.

**Comment by borsini** [comment]
Such an addition would be awesome. @ansman do you plan on opening a PR?

**Comment by ansman** [comment]
No plans until the maintainers sign off on the idea

**Comment by epool** [comment]
I did something similar using the [Showkase library](https://github.com/airbnb/Showkase) and following the [Paparazzi sample](https://github.com/airbnb/Showkase/tree/master/showkase-screenshot-testing-paparazzi-sample) as reference

```kotlin
companion object {
    @JvmStatic
    @Parameterized.Parameters(name = "{0}")
    fun parameters() = Showkase.getMetadata()
        .componentList
        .map(::ComponentTestPreview)
        .toList()
        .toTypedArray()
}

class ComponentTestPreview(
    private val showkaseBrowserComponent: ShowkaseBrowserComponent
) {
    private val showkaseComponentNameParts = showkaseBrowserComponent.componentKey.split("_")

    private val packageName: String
        get() = showkaseComponentNameParts[0]

    private val componentName: String
        get() = showkaseComponentNameParts[1]

    private val parameterizedIndex: String
        get() = showkaseComponentNameParts.getOrNull(7)?.let { "_$it" }.orEmpty()

    @Composable
    fun Content() = showkaseBrowserComponent.component()

    override fun toString(): String =
        "$packageName.$componentName$parameterizedIndex"
}
```

**Comment by VirtualParticle** [comment]
+1 for allowing a different file structure to organize screenshots. By replacing the delimiters in a few places it's possible to make it generate an output like one of these two options, whichever seems more convenient.


|![image](https://github.com/cashapp/paparazzi/assets/20135416/9599f15b-3bf6-413e-82a8-93d40f25bf5c)|![image](https://github.com/cashapp/paparazzi/assets/20135416/41febae9-7070-4f56-9fbc-7188f5e76d3b)|
|---|---|

This seems to be the same issue as #549

**Comment by leo-beliakov** [comment]
@jrodbx Any updates on this?

**Comment by matejdro** [comment]
But this workaround only works for writing? It does not specify where Paparazzi verifies images from?

**Comment by tonfido** [comment]
@matejdro exactly, but this was enough for my use-case. I guess you can still take the idea and extend paparazzi with a custom implementation.

**Comment by leofirespotter** [comment]
+1 on this as well, any way to removed the very duplicative `<packageName>.snapshotgenerator_PaparazziComposePreviewTests_previewTests` at the beginning of each file name would be great. Makes the file name too long for Windows in some cases, and also just is a bunch of noise.

**Comment by jamesonwilliams** [comment]
In my case, I had so many parameters that the screenshot generation failed, and "File name too long" was actually in the stack trace. So it wasn't just that the name became too crazy for a human to read; it was too crazy for the machine, too.

**Comment by nilsi** [comment]
We have run into similar issues where the names becomes too long on Windows. Would be great to have a way of shorten them!
